primitives/src/crh/bowe_hopwood/mod.rs

```diff
@@ -140,8 +149,9 @@ impl<G: Group, W: PedersenWindow, const N: usize> FixedLengthCRH for BoweHopwood
         ));
         let generators = Self::create_generators(rng);
         end_timer!(time);
-
-        Ok(Self::Parameters { generators, maximum_num_chunks_in_segment })
+        let parameters = Self::Parameters::from_generators(generators);
```

Comment:
Let's use the normal constructor here, otherwise we would re-execute the checks

Reply:
Ok